### ISSUE TITLE
[courier] Xenial packages built with older openssl

### ISSUE THREAD

**Issue by markc** [oerdnj/deb.sury.org#334]
I've been testing this feature of the latest 0.76.0 packages and it seems it is still not working correctly...

    courier, courier-imap: add support for TLS SNI when Courier is built with OpenSSL.

Some dialog and debugging with mrsam on the courier-users mailing-list indicates the packages were actually built with openssl 1.0.1 rather than the necessary xenial 1.0.2 that provides SNI support.

~ strings /usr/bin/couriertls | grep -i openssl
OPENSSL_1.0.0
OPENSSL_1.0.1

**Comment by markc** [comment]
Update: it seems some further digging by mrsam turns out to be a coding issue. The only way I can test this is by setting up Archlinux and using a modified AUR source package.

_I don't think that's it. I don't think I had it quite 100% right. The primary Openssl API function that's needed to implement this is SSL_set_SSL_CTX(). It does not have official documentation. SSL_get_SSL_CTX() is documented, but not SSL_set_SSL_CTX(). The only available information that could be found is in Google searches; with some hits on code samples on stackoverflow, and elsewhere.

To make a long story short, my information was incomplete, and I had to pull up and read OpenSSL sources, and attempt to figure out what's missing. I created and uploaded build 20160430, which I'm fairly certain now will work right. You're welcome to try and verify it._

http://downloads.sourceforge.net/project/courier/courier-devel/20160430/courier-0.76.0.20160430.tar.bz2
